File: Arrays/minMax.cpp
#include <bits/stdc++.h>
using namespace std;
// Question 1: find the minimum and maximum element in the array

pair<int, int> miniMaxi(vector<int> arr, int n)
{
    // optimised approach : time : O(n), space : O(1)
    int mini = INT_MAX;
    int maxi = INT_MIN;

    for (int i = 0; i < n; i++)
    {
        if (arr[i] < mini)
        {
            mini = arr[i];
        }

        if (arr[i] > maxi)
        {
            maxi = arr[i];
        }
    }
    return {maxi, mini};
}
int main()
{

    vector<int> arr = {2, 4, 1, 11, 1001};
    int n = arr.size();
    pair<int, int> ans = miniMaxi(arr, n);

    cout << "The minimum element of the array is :" << ans.second << endl;
    cout << "The maximum element of the array is :" << ans.first << endl;

    return 0;
}

File: Arrays/minMax.js
// Brute force approach
function findLargest_Smallest_Brute(arr) {
  // concept : sorting , time :O(n log n ), space : O(1)
  arr = arr.sort((a, b) => a - b);
  let n = arr.length;
  let ans = [];
  ans.push(arr[0]); // smallest element
  ans.push(arr[n - 1]); // largest element
  return ans;
}

// function for finding largset and smallest number in the array
function findLargest_Smallest(arr) {
  let ans = [];
  let mini = arr[0];
  let maxi = arr[0];
  for (let i = 0; i < arr.length; i++) {
    if (arr[i] > maxi) {
      maxi = arr[i];
    }
    if (arr[i] < mini) {
      mini = arr[i];
    }
  }
  ans.push(mini);
  ans.push(maxi);
  return ans;
}
let arr = [1, 23, 401, 100, 10];
console.log("The value of array is :" + arr);

let arr2 = [1, 2, 2, 3, 4, 5, 1, 9, 10, 10];
console.log(arr2);

let ans = findLargest_Smallest(arr2);
console.log("The largest element is " + ans[1]);
console.log("The smallest element is " + ans[0]);

let ans2 = findLargest_Smallest_Brute(arr2);
console.log("The largest element is " + ans2[1]);
console.log("The smallest element is " + ans2[0]);
